app/src/main/java/com/junkfood/seal/ui/page/settings/about/AboutPage.kt

```diff
@@ -212,22 +213,27 @@ fun AutoUpdateUnavailableDialog(onDismissRequest: () -> Unit = {}) {
     )
 
     val annotatedString = buildAnnotatedString {
-        append(text)
+        append(text.substring(0, text.indexOf(hyperLinkText)))
```

Comment:
what if the `hyperLinkText` is not at the end of the `text`?

Reply:
Since the `hyperLinkText` is being a parameter for another string there is no other way than doing a substring for putting the style and click listening in the new implementation.

In this case I believe that the current implementation is well done. Modifying the string resource is not viable. Any ideas are welcome.

Reply:
This is the only thing I think that might be okay, but tell me some other options If you are not comfortable with the changes and I'll do them. Thanks!
```kotlin
    val annotatedString = buildAnnotatedString {
        append(text.substring(0, text.indexOf(hyperLinkText)))

        withLink(
            LinkAnnotation.Clickable(
                tag = "Link to the latest app release in GitHub",
                styles = TextLinkStyles(
                    SpanStyle(
                        color = MaterialTheme.colorScheme.tertiary,
                        textDecoration = TextDecoration.Underline,
                    )
                ),
                linkInteractionListener = { _ ->
                    uriHandler.openUri("https://github.com/JunkFood02/Seal/releases/latest")
                    hapticFeedback.performHapticFeedback(HapticFeedbackType.LongPress)
                }
            )
        ) {
            append(hyperLinkText)
        }
    }
```

Reply:
not really understand what you meant on this:
1. There's an `addLink()` which is a drop-in replacement of the `addUrlAnnotation` here
2. If you'd go with `withLink()`, you should append the rest of the `text` which is **after** the `hyperLinkText` (if exist)
```kt
        append(text.substring(0, startIndex))
        //...
        append(hyperLinkText)
        //...
        append(text.substring(endIndex, text.length))
```